Framework/Core/include/Framework/AnalysisDataModel.h

```diff
@@ -63,15 +64,15 @@ DECLARE_SOA_DYNAMIC_COLUMN(Pt, pt, [](float signed1Pt) -> float { return fabs(1.
 DECLARE_SOA_DYNAMIC_COLUMN(Charge, charge, [](float signed1Pt) -> short { return (signed1Pt > 0) ? 1 : -1; });
 DECLARE_SOA_DYNAMIC_COLUMN(Px, px, [](float signed1Pt, float snp, float alpha) -> float {
   double pt = 1. / std::abs(signed1Pt);
-  double cs = std::cos(alpha);
-  double sn = std::sin(alpha);
+  float cs, sn;
+  utils::sincosf(alpha, sn, cs);
   double r = std::sqrt((1. - snp) * (1. + snp));
   return pt * (r * cs - snp * sn);
 });
 DECLARE_SOA_DYNAMIC_COLUMN(Py, py, [](float signed1Pt, float snp, float alpha) -> float {
   double pt = 1. / std::abs(signed1Pt);
-  double cs = std::cos(alpha);
-  double sn = std::sin(alpha);
+  float cs, sn;
+  utils::sincosf(alpha, sn, cs);
```

Comment:
Why do we have our own wrapper to `::sincosf`?

Reply:
Ah, ok... GNU_SOURCE

Reply:
yes, it is not portable...

Reply:
Might be related to some GPU off-loading, maybe @davidrohr or @shahor02 can comment?

Reply:
It does not work on Mac too...

Reply:
Yes, also tried that, so I think we leave it to the os::utils version, no?